src/entropy.rs

```
@@ -0,0 +1,339 @@
+//! Summary statistics (e.g. mean, variance, etc.).
+use ndarray::{Array1, ArrayBase, Data, Dimension};
+use num_traits::Float;
+
+/// Extension trait for `ArrayBase` providing methods
+/// to compute information theory quantities
+/// (e.g. entropy, Kullback–Leibler divergence, etc.).
+pub trait EntropyExt<A, S, D>
+    where
+        S: Data<Elem = A>,
+        D: Dimension,
+{
+    /// Computes the [entropy] *S* of the array values, defined as
+    ///
+    /// ```text
+    ///       n
+    /// S = - ∑ xᵢ ln(xᵢ)
+    ///      i=1
+    /// ```
+    ///
+    /// If the array is empty, `None` is returned.
+    ///
+    /// **Panics** if any element in the array is negative.
```

SciPy [defines the entropy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.entr.html#scipy.special.entr) for a negative element as negative infinity instead of panicking. This is nice because it avoids panics, but I'm not sure whether defining it as negative infinity really makes sense, though. Thoughts?

---

Mathematically, it does not and it can turn out into a silent failure. I'd rather be vocal and upfront about invalid values (return an error/panic).

---

Okay, sounds good to me.